### PR TITLE
[unwind] reduce frame size of mono_unwind_frame ()

### PR DESCRIPTION
**Pull request by lewurm**
on arm32 it goes down from 5120 bytes to 676 bytes in terms of frame size.

/cc @vargaz

### CODE REVIEW

File: mono/mini/unwind.c [hunk 1]
```diff
@@ -46,19 +46,19 @@ static int unwind_info_size;
 
 #ifdef TARGET_AMD64
 static int map_hw_reg_to_dwarf_reg [] = { 0, 2, 1, 3, 7, 6, 4, 5, 8, 9, 10, 11, 12, 13, 14, 15, 16 };
-#define NUM_REGS AMD64_NREG
+#define DWARF_REGS AMD64_NREG
 #define DWARF_DATA_ALIGN (-8)
```
Comment (vargaz) [inline mono/mini/unwind.c]: This should be NUM_DWARF_REGS or NDWARF_REGS.

File: mono/mini/unwind.c [hunk 2]
```diff
@@ -100,23 +100,25 @@ static int map_hw_reg_to_dwarf_reg [32] = {
...
 #else
 static int map_hw_reg_to_dwarf_reg [16];
-#define NUM_REGS 16
+#define DWARF_REGS 16
 #define DWARF_DATA_ALIGN 0
 #define DWARF_PC_REG -1
 #endif
 
+#define HW_REGS (sizeof (map_hw_reg_to_dwarf_reg) / sizeof (int))
+
```
Comment (vargaz) [inline mono/mini/unwind.c]: Ditto.

**Comment by vargaz** [comment]
Looks ok otherwise.

**Comment by lewurm** [comment]
thanks, updated.  @monojenkins merge

**Pull request merged**